### DISCUSSION TITLE
Cloning worksheets at specific positions

### DISCUSSION THREAD

**Discussion by Jeff-White-AZ** [JanMarvin/openxlsx2#1089] [Q&A] [answered]
I see there is a warning (understandably) that the cloning function may not always work as expected. In my case, I want to replace a worksheet in a data template file with a revised worksheet:
target_wb$clone_worksheet(old = updated_sheet, new = updated_sheet, from = wb_template)
But I have two issues:
1. If I don't delete the existing 'updated_sheet' in 'target_wb', the sheet is appended and on opening in Excel, an error is reported and the contents of the sheet are in a sheet at the end of the existing sheets. 
2. If I do delete the existing 'updated_sheet' in 'target_wb', the new 'updated_sheet' appears as the last sheet in 'target_wb'.
So two suggestions for clone_worksheet() might be:
A. Either prohibit overwriting of existing sheets or add an option for overwriting.
B. Provide a means to specify the position of a sheet cloned to a workbook.

**Comment by Jeff-White-AZ** [comment]
I see that for my suggested option two, there is a way to write code to re-order the sheets using wb$set_order().
I will use this for now but simply specifying the sheet position seems more transparent and similar to Excel functionality with copy/move.

**Reply by JanMarvin** [reply, marked as answer]
Hi @Jeff-White-AZ , 

cloning is a nasty thing. So many things can go wrong and have to be taken care of. It has come a long way, but it was not a "walk in the park" to say the least and it definitely isn't perfect, hence the warning. It isn't likely that I'm going to tinker with it any time soon, therefore a custom function that does what you do seems like the best solution at hand.

Something like this:
* check for matching spreadsheet names
* get the position of the matching spreadsheet name in the workbook you want to clone into
* remove the matching spreadsheet in the workbook you want to clone into
* clone the worksheet
* update the sheet order

**Reply by Jeff-White-AZ** [reply]
Thanks for the suggestion, I will incorporate your five steps in a function replace_sheet().
